### PR TITLE
Add docstrings for user-visible classes, functions, and constants

### PR DESCRIPTION
**Pull request by krishnangovindraj**

### CODE REVIEW

File: python/typedb/api/query/query_manager.py [hunk 1]
```diff
@@ -37,41 +37,172 @@ class QueryManager(ABC):
...
+
+        :param query:   The TypeQL Match (Get) query to be executed
+        :param options: Specify query options
+        :return:
+
+        Examples
+        --------
+        ::
+
+            transaction.query().match(query, options)
```
Comment (dmikhalin) [inline python/typedb/api/query/query_manager.py]: `query` is a property now.

File: python/typedb/api/logic/logic_manager.py [hunk 2]
```diff
@@ -29,15 +29,55 @@
...
+        Retrieves the Rule that has the given label.
+
+        :param label: The label of the Rule to create or retrieve
+        :return:
+
+        Examples:
+        ---------
+        ::
+
+            transaction.logic().get_rule(label)
```
Comment (dmikhalin) [inline python/typedb/api/logic/logic_manager.py]: `logic` is a property now.

File: python/typedb/api/user/user.py [hunk 3]
```diff
@@ -28,54 +28,133 @@ class User(ABC):
...
+        Checks if a user with the given name exists.
+
+        :param username: The user name to be checked
+        :return:
+
+        Examples:
+        ---------
+        ::
+
+           driver.users().contains(username)
```
Comment (dmikhalin) [inline python/typedb/api/user/user.py]: User manager is also retrieved by property.

**Review by dmikhalin — CHANGES_REQUESTED** [review]
We need to remove some parentheses from examples, because we transformed some getters to properties.

File: python/typedb/api/user/user.py (hunk 3 above)
Comment (krishnangovindraj) [reply python/typedb/api/user/user.py]: Updated.

File: python/typedb/api/logic/logic_manager.py (hunk 2 above)
Comment (krishnangovindraj) [reply python/typedb/api/logic/logic_manager.py]: Updated.

File: python/typedb/api/query/query_manager.py (hunk 1 above)
Comment (krishnangovindraj) [reply python/typedb/api/query/query_manager.py]: Updated.

**Review by dmikhalin — APPROVED** [review]
Thanks for your help!

**Pull request merged**